### PR TITLE
Allow file protocol during source-build clone.

### PR DESCRIPTION
**Pull request by crummel**
Fixes https://github.com/dotnet/source-build/issues/3111.

### CODE REVIEW

**Review by MichaelSimons — DISMISSED** [review]
Will you also backport this to 6.0 and 7.0?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/git-clone-to-dir.sh [hunk 1]
```diff
@@ -124,12 +124,12 @@ fi
       # * '--no-local' allows a shallow clone from a git dir on the same filesystem. This means the
       #   clone will not use hard links and takes up more space. However, since we're doing a shallow
       #   clone anyway, the difference is probably not significant. (This has not been measured.)
-      git clone --depth=1 --no-local --no-checkout "$sourceDir" "$destDir"
+      git clone -c protocol.file.allow=always --depth=1 --no-local --no-checkout "$sourceDir" "$destDir"
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/git-clone-to-dir.sh:127]: Can you add a comment to indicate the purpose of this?

Comment (crummel) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/git-clone-to-dir.sh:129]: Added as suggested.

**Review by mmitche — APPROVED** [review]

**Comment by crummel** [comment]
> Will you also backport this to 6.0 and 7.0?

Opened https://github.com/dotnet/arcade/pull/12298 and https://github.com/dotnet/arcade/pull/12299.

**Review by dougbu — APPROVED** [review]

**Pull request merged**